gm4_metallurgy/data/metallurgy/functions/casting/check_mould_creation.mcfunction

```diff
@@ -0,0 +1,3 @@
+# @s = obsidian block with count of 1 on top of sand
+# run from main
+execute align xyz unless entity @e[type=vex,tag=gm4_sand_ring,dx=0,dy=-1,dz=0] if entity @e[type=item,dx=0,dy=0,dz=0,nbt={Item:{id:"minecraft:clay_ball",Count:1b},OnGround:1b}] run function metallurgy:casting/create_mould
```

Comment:
I don't see why this is it's own function and not appended to the execute in `metallurgy/functions/main.mcfunction` but that's fine with me.

Reply:
execute makes it list of entities before calling the first function, so if this execute was chained in `main`, it could remove two obsidian items for one clay.

Reply:
Alright!